### ISSUE TITLE
No checks with typed context

### ISSUE THREAD

**Issue by StarpTech** [avajs/eslint-plugin-ava#331]
No error is detected but works with `import test from 'ava'`

```ts
import anyTest, { ExecutionContext, TestInterface } from 'ava';
const test = anyTest as TestInterface<TestContext>;
test.only('Should create a new profile',
```

```js
module.exports = {
	parser: '@typescript-eslint/parser', // Specifies the ESLint parser
	parserOptions: {
		ecmaVersion: 2020, // Allows for the parsing of modern ECMAScript features
		sourceType: 'module', // Allows for the use of imports
	},
	extends: [
		'plugin:@typescript-eslint/recommended', // Uses the recommended rules from the @typescript-eslint/eslint-plugin
		'plugin:ava/recommended',
		'plugin:prettier/recommended', // Enables eslint-plugin-prettier and eslint-config-prettier. This will display prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
	],
	plugins: ['import']
};

```

**Comment by Darkle** [comment]
I get the same issue. 
Running eslint v8.32.0 with eslint-plugin-ava v14.0.0

**Comment by Darkle** [comment]
I created a test repo: https://github.com/Darkle/ava-eslint-test
